CryptoYa: Create ExchangeRate per exchange

alvasw opened the pull request:
CryptoYa queries 20 exchanges and returns their data. It makes more sense to pass the data from each exchange separately to the ExchangeRateService. The ExchangeRateService aggregates all exchange rates later.

ghost commented:
ACK - Awesome!

    INFO  b.p.s.ExchangeRateService: CryptoYa: ripioexchange ARS outlier price removed:3.447988443E7, lower/upper bounds:1.827556019215259E7/1.98456237E7, consensus price:1.943283936853435E7

alvasw committed:
CryptoYa: Create ExchangeRate per exchange

CryptoYa queries 20 exchanges and returns their data. It makes more
sense to pass the data from each exchange separately to the
ExchangeRateService. The ExchangeRateService aggregates all exchange
rates later.

alvasw committed:
ExchangeRateService: Filter by provider names with startsWith

alvasw commented on src/main/java/bisq/price/spot/ExchangeRateService.java:
```diff
@@ -237,7 +237,7 @@ private Map<String, Object> getMetadata(ExchangeRateProvider provider) {
     private long getTimestamp(ExchangeRateProvider provider, Set<ExchangeRate> exchangeRates) {
         return exchangeRates.stream()
-                .filter(e -> provider.getName().equals(e.getProvider()))
+                .filter(e -> provider.getName().startsWith(e.getProvider()))
```
Thank you and sorry about that! :-)

gabernard reviewed (approved):
utACK based on https://github.com/bisq-network/bisq-pricenode/pull/30#pullrequestreview-1630098297.
